lms/djangoapps/discussion_api/api.py

```diff
@@ -901,7 +913,9 @@ def update_thread(request, thread_id, update_data):
         The updated thread; see discussion_api.views.ThreadViewSet for more
         detail.
     """
-    cc_thread, context = _get_thread_and_context(request, thread_id)
+    cc_thread, context = _get_thread_and_context(
+        request, thread_id, retrieve_kwargs={"user_id": unicode(request.user.id)}
```

Comment:
Same as my [comment above](https://github.com/edx/edx-platform/pull/13431#discussion-diff-78182016R759).

When `read` is true `unread_comment_count` is always 0.